### PR TITLE
API-395: Add route to get list of product models

### PR DESCRIPTION
**Pull request by willy-ahva**

### CODE REVIEW

File: content/akeneo-web-api.yaml [hunk 1]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                    type: object
+                    properties:
+                      _items:
+                        type: array
+                        items:
+                          $ref: '#/definitions/ProductModelList'
+          examples: {
+            "_links": {
+              "self": {
+                "href": "http://localhost/api/rest/v1/product-models?pagination_type=search_after&limit=3&search_after=qg%3D%3D"
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Remove `localhost`and use `https://demo.akeneo.com`

File: content/akeneo-web-api.yaml [hunk 2]
```diff
@@ -602,6 +602,184 @@ paths:
...
+              },
+              "first": {
+                "href": "http://localhost/api/rest/v1/product-models?pagination_type=search_after&limit=3"
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Idem

File: content/akeneo-web-api.yaml [hunk 3]
```diff
@@ -602,6 +602,184 @@ paths:
...
+              },
+              "next": {
+                "href": "http://localhost/api/rest/v1/product-models?pagination_type=search_after&limit=3&search_after=rw%3D%3D"
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Idem

File: content/akeneo-web-api.yaml [hunk 4]
```diff
@@ -602,6 +602,184 @@ paths:
...
+              }
+            },
+            "_embedded": {
+              "items": [
+                {
+                  "_links": {
+                    "self": {
+                      "href": "http:\/\/localhost\/api\/rest\/v1\/product-models\/shoes"
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Idem

File: content/akeneo-web-api.yaml [hunk 5]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                    }
+                  },
+                  "code": "shoes",
+                  "family_variant": "familyVariantA1",
+                  "parent": null,
+                  "categories": ["clothing", "shoes"],
+                  "values": {
+                    "a_price": [
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Give real attribute name. Here `price` is  enough.

File: content/akeneo-web-api.yaml [hunk 6]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                        "scope": null,
+                        "data": [
+                          {
+                            "amount": "50.00",
+                            "currency": "EUR"
+                          }
+                        ]
+                      }
+                    ],
+                    "a_number_float": [
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Idem. A real attribute code name.

File: content/akeneo-web-api.yaml [hunk 7]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                      {
+                        "locale": null,
+                        "scope": null,
+                        "data": "12.5000"
+                      }
+                    ],
+                    "a_localized_and_scopable_text_area": [
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: `description` would be better here as an attribute code.

File: content/akeneo-web-api.yaml [hunk 8]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                      }
+                    ]
+                  },
+                  "created": "2017-10-04T18:04:10+02:00",
+                  "updated": "2017-10-04T18:04:10+02:00"
+                },
+                {
+                  "_links": {
+                    "self": {
+                      "href": "http:\/\/localhost\/api\/rest\/v1\/product-models\/tshirt"
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Same comments as above for this whole product-model (url + attributes code)

File: content/akeneo-web-api.yaml [hunk 9]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                      }
+                    ]
+                  },
+                  "created": "2017-10-04T18:04:10+02:00",
+                  "updated": "2017-10-04T18:04:10+02:00"
+                },
+                {
+                  "_links": {
+                    "self": {
+                      "href": "http:\/\/localhost\/api\/rest\/v1\/product-models\/trousers"
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: Idem

File: content/akeneo-web-api.yaml [hunk 10]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                  "updated": "2017-10-04T18:04:10+02:00"
+                }
+              ]
+            }
+          }
+        400:
+          $ref: "#/responses/400Error"
+        401:
+          $ref: "#/responses/401Error"
+        406:
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: It's missing the 422 error, here.

File: content/akeneo-web-api.yaml [hunk 11]
```diff
@@ -619,12 +797,14 @@ paths:
           x-details: Returns the content of the product model in JSON standard format.
           schema:
             $ref: '#/definitions/ProductModel'
+        400:
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: There can't be any 400 in this request since you don't send any body.

File: content/akeneo-web-api.yaml [hunk 12]
```diff
@@ -602,6 +602,184 @@ paths:
...
+                      }
+                    ]
+                  },
+                  "created": "2017-10-04T18:04:10+02:00",
+                  "updated": "2017-10-04T18:04:10+02:00"
+                }
+              ]
+            }
+          }
+        400:
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: No 400 on a GET.

File: content/akeneo-web-api.yaml (hunk 10 above)
Comment (willy-ahva) [reply content/akeneo-web-api.yaml]: There is no search available so 422 should never happen

Comment (NolwennP) [reply content/akeneo-web-api.yaml]: See API-32. You can have 422 if there are errors on the query parameters such as `page`.

File: content/akeneo-web-api.yaml [hunk 13]
```diff
@@ -602,6 +602,161 @@ paths:
...
+                    type: object
+                    properties:
+                      _items:
+                        type: array
+                        items:
+                          $ref: '#/definitions/ProductModelList'
+          examples: {
+            "_links": {
+              "self": {
+                "href": "http://demo.akeneo.com/api/rest/v1/product-models?pagination_type=search_after&limit=3&search_after=qg%3D%3D"
```
Comment (NolwennP) [inline content/akeneo-web-api.yaml]: There shouldn't be a `search_after` parameter here, I think.

Comment (NolwennP) [reply content/akeneo-web-api.yaml]: Nevermind!

**Pull request merged**